astropy/units/tests/test_units.py

```diff
@@ -292,62 +292,78 @@ def test_empty_compose():
         composed = u.m.compose(units=[])
 
 
-def test_compose_roundtrip():
-    def _test_compose_roundtrip(unit):
-        composed_list = unit.decompose().compose()
-        found = False
-        for composed in composed_list:
-            if len(composed.bases):
-                if composed.bases[0] is unit:
-                    found = True
-                    break
-            elif len(unit.bases) == 0:
+def _unit_as_str(unit):
+    # This function serves two purposes - it is used to sort the units to
+    # test alphabetically, and it is also use to allow pytest to show the unit
+    # in the [] when running the parametrized tests.
+    return str(unit)
+    
+    
+# We use a set to make sure we don't have any duplicates.
+COMPOSE_ROUNDTRIP = set()
+for val in u.__dict__.values():
+    if (isinstance(val, u.UnitBase) and
+            not isinstance(val, u.PrefixUnit)):
+        COMPOSE_ROUNDTRIP.add(val)
+
+
+@pytest.mark.parametrize('unit', sorted(COMPOSE_ROUNDTRIP, key=_unit_as_str), ids=_unit_as_str)
```

Comment:
Oh, I never scroll down far enough in the `pytest` doc to know that you can pass in IDs like this. Nice! :+1: